pkg/sdn/api/types.go

```diff
@@ -60,3 +60,33 @@ type NetNamespaceList struct {
 	unversioned.ListMeta
 	Items []NetNamespace
 }
+
+// EgressFirewallPolicy gives the type of an EgressFirewallRule
+type EgressFirewallPolicy string
+
+const (
+	EgressFirewallPolicyAllow EgressFirewallPolicy = "allow"
+	EgressFirewallPolicyDeny  EgressFirewallPolicy = "deny"
+)
+
+// EgressFirewallRule contains a single outgoing traffic firewall rule
+type EgressFirewallRule struct {
+	Policy EgressFirewallPolicy
+	From   string // Namespace name or empty
+	To     string // CIDR range
+}
+
+// EgressFirewall provides a list of firewall rules on outgoing traffic
+type EgressFirewall struct {
+	unversioned.TypeMeta
+	kapi.ObjectMeta
+
+	Rules []EgressFirewallRule
```

Comment:
Should be in a spec (same as network policy) and Rules should be called `Egress []EgressFirewallRule`